File: fittness-app/parent-module/selfCoach/src/main/java/tudor/work/model/Achievement.java
package tudor.work.model;


import com.fasterxml.jackson.annotation.JsonBackReference;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table
public class Achievement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long name;

    private Long description;

    private Integer numberOfPoints;

    private String achievementPicturePath;

    @ManyToMany(mappedBy = "achievements",cascade = CascadeType.ALL)
    @JsonBackReference
    private Set<User> users;

}

File: fittnes_frontend/lib/models/achievement.dart
class Achievement {
  final int id;
  final String name;
  final String description;
  final double numberOfPoints;
  final String achievementPicturePath;

  Achievement(
      {required this.id,
      required this.name,
      required this.description,
      required this.numberOfPoints,
      required this.achievementPicturePath});

  factory Achievement.fromJson(Map<String, dynamic> json) {
    return Achievement(
        id: json['id'],
        name: json['name'],
        description: json['description'],
        numberOfPoints: json['numberOfPoints'],
        achievementPicturePath: json['achievementPicturePath']?? '');
  }

  
}
